Sometimes sops store broken data key (invalid key size).

After a few key rotations sops can leave encrypted file in a broken state.

Steps to reproduce:
1) Generate new encrypted file with single PGP key:
`sops -pgp <fingerprint> tmp.yml`
2) Launch a loop that constantly rotates data key: for bash
`while [[ $? == 0 ]]; do; sops -r -i tmp.yml; done`
3) Wait until the loop fails with error:

```
[PGP]	 INFO[0000] Decryption succeeded                          fingerprint=<fingerprint>
[SOPS]	 INFO[0000] Data key recovered successfully              
Error decrypting tree: Error walking tree: Could not decrypt value: crypto/aes: invalid key size 31
```

It usually takes 5-10 cycles to break the file.

This behaviour is observed in version 3.0.0 (in linux version downloaded from [releases](https://github.com/mozilla/sops/releases/download/3.0.0/sops-3.0.0.linux), as well as in darwin version installed by `brew install sops`).

PGP agent:
```
$ gpg --version
gpg (GnuPG) 2.1.21
libgcrypt 1.7.7
```

Example of decrypted data key from broken file:

```
$ cat tmp.yml | ruby -ryaml -e 'puts YAML.load(STDIN.read)["sops"]["pgp"][0]["enc"]' | gpg -d | hexdump -C
00000000  9c 74 86 a4 54 81 72 e1  92 9f f5 80 1d d0 c4 92  |.t..T.r.........|
00000010  9e 39 b7 dc 00 93 5f d7  3e 15 3f 78 13 69 3d     |.9...._.>.?x.i=|
0000001f
```

What's interesting is that the automated quickcheck-like test we have for this does fail for me consistently:
```
❯ go test go.mozilla.org/sops/pgp
[PGP]	 time="2017-11-27T19:33:05+01:00" level=info msg="Encryption succeeded" fingerprint=1022470DE3F0BC54BC6AB62DE05550BC07FB1A0A
[PGP]	 time="2017-11-27T19:33:05+01:00" level=info msg="Decryption succeeded" fingerprint=1022470DE3F0BC54BC6AB62DE05550BC07FB1A0A
--- FAIL: TestPGP (1.79s)
	keysource_test.go:29: #1: failed on input []byte{0x3b, 0xc6, 0xd4, 0x87, 0x63, 0xfa, 0x2, 0x43, 0x85, 0xc4, 0x45, 0x92, 0x42, 0x93, 0x7, 0x5c, 0x9c, 0x9c, 0xd, 0xba, 0x1b, 0xe8, 0xc1, 0x3c, 0xb2, 0xd7, 0x12, 0xf7, 0x16, 0xee, 0x86, 0x7f, 0x57, 0xb7, 0xed, 0x5, 0xa5, 0x2e, 0x25}
FAIL
FAIL	go.mozilla.org/sops/pgp	1.963s
```

, but it seems to [work fine on CI](https://travis-ci.org/mozilla/sops/builds/304656807#L664).

--- a/pgp/keysource.go
+++ b/pgp/keysource.go
@@ -127,7 +127,7 @@ func (key *MasterKey) encryptWithCryptoOpenPGP(dataKey []byte) error {
 	if err != nil {
 		return err
 	}
-	plaintextbuf, err := openpgp.Encrypt(armorbuf, []*openpgp.Entity{&entity}, nil, nil, nil)
+	plaintextbuf, err := openpgp.Encrypt(armorbuf, []*openpgp.Entity{&entity}, nil, &openpgp.FileHints{IsBinary: true}, nil)
 	if err != nil {
 		return err
 	}

